### PR TITLE
Documentation: add guide for pysimCoder integration with NuttX

### PR DESCRIPTION
**Pull request by michallenc**
## Summary
This commit adds documentation for compiling NuttX for pysimCoder and then running pysimCoder designed applications on NuttX. PysimCoder is a Rapid Control Application Development Tool that can transfer block diagrams into C code and can be used in real time control applications.

## Impact
Documentation only.

### CODE REVIEW

File: Documentation/guides/pysimcoder.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
+.. include:: /substitutions.rst
+.. _pysimcoder:
+
+pysimCoder integration with NuttX
+=================================
+
+`PysimCoder <https://github.com/robertobucher/pysimCoder>`_ is as an open source
+Rapid Control Application Development Tool which is abe to transfer block diagrams into C code.
```
Comment (Ouss4) [inline Documentation/guides/pysimcoder.rst:8]:
```suggestion
Rapid Control Application Development Tool which is able to transfer block diagrams into C code.
```

File: Documentation/guides/pysimcoder.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+Combined with NuttX, it can be used in real time control application and as an alternative to
+expensive licensed programs and prototyping platforms. Example of `DC motor control application
+<https://www.youtube.com/watch?v=6HlGk3ecPNQ>`_ with PID controller and block blocks for encoder,
```
Comment (Ouss4) [inline Documentation/guides/pysimcoder.rst:11]: double word "block(s)"?

File: Documentation/guides/pysimcoder.rst [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+ENC
+GPIO
+PWM         Multichannel support
+UART        Serial output
+Sensors     Basic support for DHTXX
+TCP
+UDP
+==========  =======================
+
+Please note that the actual support for NuttX peripherals can be wider that it is mentioned here
```
Comment (Ouss4) [inline Documentation/guides/pysimcoder.rst:38]:
```suggestion
Please note that the actual support for NuttX peripherals can be wider than what is mentioned here
```

File: Documentation/guides/pysimcoder.rst [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+in case this documentation was not updated when new fuctionalities were added to pysimCoder.
+
+NuttX Configuration
+-------------------
+
+Several configuration options are neccessary to be set in order to succesfully compile pysimCoder
+with NuttX. The list is following:
```
Comment (Ouss4) [inline Documentation/guides/pysimcoder.rst:45]:
```suggestion
with NuttX. The list is the following:
```

**Comment by michallenc** [comment]
Hi @Ouss4, thanks.

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**